### PR TITLE
add arm64 support

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
I've been sitting on this change for a while: figure it's time to share it upstream. Note that it requires bumping the mattermost-server `go.mod` version because of support we added in the `model` struct to store an arbitrary dictionary of platforms to binaries vs. just the ones previously hard-coded.

#### Ticket Link
None.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -61,11 +61,13 @@ ifneq ($(HAS_SERVER),)
 ifeq ($(MM_DEBUG),)
 	cd server && env GOOS=linux GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-linux-amd64;
 	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-darwin-amd64;
+	cd server && env GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-darwin-arm64.exe;
```
Comment (hanzei) [inline Makefile:64]: Why is common windows file extension `.exe` used here?

File: Makefile [hunk 2]
```diff
@@ -61,11 +61,13 @@ ifneq ($(HAS_SERVER),)
...
 	cd server && env GOOS=windows GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-windows-amd64.exe;
 else
 	$(info DEBUG mode is on; to disable, unset MM_DEBUG)
 
 	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-darwin-amd64;
+	cd server && env GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-windows-arm64.exe;
```
Comment (hanzei) [inline Makefile:70]: Should this be
```suggestion
	cd server && env GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) -gcflags "all=-N -l" -o dist/plugin-darwin-arm64.exe;
```
?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for adding this. :+1: Two questions below

File: Makefile (hunk 1 above)
Comment (lieut-data) [reply Makefile]: Ugh, I know now why I was still sitting on this -- this was just one commit from a two parter that I forgot to cherry-pick over here :) -- should be fixed now!

File: Makefile (hunk 2 above)
Comment (lieut-data) [reply Makefile]: Yeah, and this was one just wrong on my part :p -- thanks for the careful eye!

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by larkox — APPROVED** [review]

**Pull request merged**